SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "ReferenceCountedLockingPolicy.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/AuthorityIntent.h"
+
+#include "GameFramework/Actor.h"
+
+DEFINE_LOG_CATEGORY(LogReferenceCountedLockingPolicy);
+
+ActorLockToken UReferenceCountedLockingPolicy::AcquireLock(const AActor* Actor, FString DebugString)
+{
+	check(Actor != nullptr);
+	if (!CanAcquireLock(Actor)) {
+		UE_LOG(LogReferenceCountedLockingPolicy, Error, TEXT("Called AcquireLock when CanAcquireLock returned false. Actor: %s."), *Actor->GetName());
+		return SpatialConstants::INVALID_ACTOR_LOCK_TOKEN;
+	}
+	++ActorToReferenceCount.FindOrAdd(Actor);
+	UE_LOG(LogReferenceCountedLockingPolicy, Log, TEXT("Acquiring migration lock. "
+		"Actor: %s. Lock name: %s. Token %d: Locks held: %d."), *Actor->GetName(), *DebugString, NextToken, *ActorToReferenceCount.Find(Actor));
+	TokenToNameAndActor.Emplace(NextToken, LockNameAndActor{ MoveTemp(DebugString), Actor });
```

Comment:
total nit, but I'd rather pass a const ref to this function, and just copy the object here if we need it.

Reply:
For a sink I'd rather pass by value. For a pass by ref you always do one copy, for a pass by value you do a copy and a move or possibly just a move. But the type is cheap to move.
If you have a strong preference then sure but I disagree.

Reply:
We discussed offline: normal strings would be literals, FStrings are array wrappers so always pass by const ref where possible, however, using const ref it doesn't look like you can have a default param so in this case Mike's fine to leave as is.

Reply:
I'm not sure I follow what you are saying. I no change is needed then we don't need to discuss this in detail here but we might want to discuss this for the wider question of our style guide. 

>"FStrings are array wrappers so always pass by const ref where possible" 
const ref should be the default for most things but nothing I said above changes because FString is secretly a TArray.
The general choices are const ref only (always pass a pointer then take a copy),
const ref and rvalue ref so that a moved from string doesn't do a copy,
pass by value so that you don't need two functions but can still take advantage of moved from values (and literals get copy elided so there is only one move),
or templates and forwarding. 

In this case I prefer pass by value because we expect these to be moved from or literals a lot of the time and doesn't seem important enough worth having multiple overloads for. And moving an array is fairly cheap. If we really care about performance I'd recommend using two functions.
Passing a literal to a const FString ref (I think) makes a temporary that needs to be copied from and then destroyed.